src/components/avatars/OcAvatar.spec.js

```diff
@@ -0,0 +1,100 @@
+import { mount } from "@vue/test-utils"
+import OcAvatar from "./OcAvatar.vue"
+
+describe("OcAvatar", () => {
+  const selectors = {
+    initials: ".avatarInitials",
+  }
+  function getWrapperWithProps(props) {
+    return mount(OcAvatar, {
+      propsData: {
+        ...props,
+      },
+    })
+  }
+  describe("prop value", () => {
+    describe("when src is set", () => {
+      let wrapper
+      beforeEach(() => {
+        wrapper = getWrapperWithProps({
+          src: "http://some-image.jpg",
+        })
+      })
+      it("should render oc image", () => {
+        const imageElement = wrapper.find("img")
+        expect(imageElement.exists()).toBeTruthy()
+        expect(imageElement.attributes("src")).toBe("http://some-image.jpg")
+      })
+      it("should not render user initial", () => {
+        expect(wrapper.find(selectors.initials).exists()).toBeFalsy()
+      })
+      it("should not render background", () => {
+        expect(wrapper.attributes("style")).not.toContain("background-color: ")
+      })
+    })
+    describe("when username is set", () => {
+      it.each`
+        username                   | expected_initial
+        ${"test"}                  | ${"T"}
+        ${"Test"}                  | ${"T"}
+        ${"1Test"}                 | ${"1"}
+        ${"test user"}             | ${"TU"}
+        ${"test User"}             | ${"TU"}
+        ${"Test User"}             | ${"TU"}
+        ${"1234 5678"}             | ${"15"}
+        ${"test-user"}             | ${"TU"}
+        ${"test -user"}            | ${"TU"}
+        ${"test - user"}           | ${"TU"}
+        ${"test user one"}         | ${"TUO"}
+        ${"test-user-one"}         | ${"TUO"}
+        ${"test user one primary"} | ${"TUO"}
+        ${"test User One Primary"} | ${"UOP"}
+        ${"testUser One Primary"}  | ${"TOP"}
+      `("should render user initials", ({ username, expected_initial }) => {
+        const wrapper = getWrapperWithProps({
+          userName: username,
+        })
+        const userInitialElement = wrapper.find(selectors.initials)
+        expect(userInitialElement.exists()).toBeTruthy()
+        expect(userInitialElement.text()).toBe(expected_initial)
+      })
+    })
+    describe("width", () => {
+      let wrapper
+      beforeEach(() => {
+        wrapper = getWrapperWithProps({
+          width: 100,
+        })
+      })
+      it("should set width and height of the avatar wrapper", () => {
+        expect(wrapper.attributes("style")).toContain("width: 100px; height: 100px;")
+      })
+      it("should determine font style", async () => {
+        expect(wrapper.attributes("style")).toContain("line-height: 105px;")
+        expect(wrapper.attributes("style")).toContain(
+          "font-size: 40px; font-family: Helvetica, Arial, sans-serif;"
+        )
```

Comment:
that feels like testing too much implementation details

Reply:
styles are calculated using `height` inside the component, so I added the tests for it too.

Reply:
Good point, you are right :+1: 
lets then reduce it to the calculated values and leave `font-family` alone          `expect(wrapper.attributes("style")).toContain("font-size: 40px")`